src/lib/universal-client.js

```diff
@@ -0,0 +1,638 @@
+         * backend and forwards to the configurated transport handleSession
+         * function
+         * 
+         * @param json
+         */
+        this.handleLogin = function(json) {
+          // read backend configuration if send by backend
+          if (json.c) {
+            $.extend(this.config, json.c);
+            this.checkSettings();
+          }
+          // bind context object (this) to the handleSession function
+          var bound = this.transport[this.config.transport].handleSession
+              .bind(this.transport[this.config.transport]);
+          // call the bound function
+          bound(json);
+        };
+
+        /**
+         * This function stops an ongoing connection
+         * 
+         * @method stop
+         */
+        this.stop = function() {
+          this.running = false;
+          if (this.transport[this.config.transport].abort) {
+            this.transport[this.config.transport].abort();
+          }
+          // alert('this.stop');
+        };
+
+        this.transport = {
+          'long-polling' : {
+            doRestart : false, // are we currently in a restart, e.g. due to
+                                // the watchdog
+            xhr : false, // the ongoing AJAX request
+            watchdogTimer : 5, // in Seconds - the alive check interval of the
+                                // watchdog
+            maxConnectionAge : 60, // in Seconds - restart if last read is
+                                    // older
+            maxDataAge : 3200, // in Seconds - reload all data when last
+                                // successful read is older
+            // (should be faster than the index overflow at max data rate, i.e.
+            // 2^16 @ 20 tps for KNX TP)
+            lastIndex : -1, // index returned by the last request
+            retryCounter : 0, // count number of retries (reset with each valid
+                              // response)
+
+            init : function() {
+              this.watchdog();
+            },
+
+            watchdog : function() {
+              var last = new Date();
+              var hardLast = last;
+              var aliveCheckFunction = function() {
+                var now = new Date();
+                if (now - last < this.maxConnectionAge * 1000)
+                  return;
+                if (now - hardLast > this.maxDataAge * 1000)
+                  thislastIndex = -1; // reload all data
+                this.restart();
+                last = now;
+              };
+              setInterval(aliveCheckFunction, this.watchdogTimer * 1000);
+              return {
+                ping : function() {
+                  // delete last;
+                  last = new Date();
+                  if (!this.doRestart) {
+                    // delete hardLast;
+                    hardLast = last;
+                  }
+                }
+              };
+            },
+
```

Comment:
To get the watchdog to run (see error on line 371), I had to change this block to:

    watchdog : (function() {
      var last = new Date();
      var hardLast = last;
      var aliveCheckFunction = function() {
        var now = new Date();
        if (now - last < thisCometVisu.transport['long-polling'].maxConnectionAge * 1000)
          return;
        if (now - hardLast > thisCometVisu.transport['long-polling'].maxDataAge * 1000)
          thislastIndex = -1; // reload all data
        thisCometVisu.transport['long-polling'].restart();
        last = now;
      };
      setInterval(aliveCheckFunction, 5 * 1000);
      return {
        ping : function() {
          // delete last;
          last = new Date();
          if (!thisCometVisu.transport['long-polling'].doRestart) {
            // delete hardLast;
            hardLast = last;
          }
        }
      };
    })(),

but you see it's quite ugly as by using the `{ key1: value, key2: value }` notation in this case doesn't allow direct access to the other values, especially not during setup.

I guess it's better to change that to a real JS Object with constructor and local variables as there's no need to access them from the outside.

(Note: the `cometvisu-client.js` was written by me on a time where my understanding of JS wasn't as deep as it is today. So there are a few places that should be a bit different when modern JS is used. There's no need to copy the bad parts :) )